src/ElectronicBankStatement.php

```diff
@@ -0,0 +1,215 @@
+     *
+     * @var ?string
+     */
+    private $iban;
+
+    /**
+     * Code of the corresponding bank book. Either account or iban or code should be set.
+     *
+     * @var ?string
+     */
+    private $code;
+
+    /**
+     * Bank statement date. Set to the current date when left empty.
+     *
+     * @var \DateTimeInterface
+     */
+    private $date;
+
+    /**
+     * Currency code. Set to the currency of the corresponding bank day book when left empty.
+     *
+     * @var Currency
+     */
+    private $currency;
+
+    /**
+     * Number of the bank statement. When left empty, last available bank statement number increased by one.
+     *
+     * @var int
+     */
+    private $statementnumber;
+
+    /**
+     * Optional. Office in which the bank statement should be imported.
+     *
+     * @var Office
+     */
+    private $office;
+
+    /**
+     * Opening balance. If not provided, the opening balance will be based on the previous bank statement.
+     *
+     * @var Money
+     */
+    private $startvalue;
+
+    /**
+     * Closing balance. If not provided, the closing balance will be based on the opening balance and the total amount of the transactions.
+     *
+     * @var Money
+     */
+    private $closevalue;
+
+    public function getAccount(): ?string
+    {
+        return $this->account;
+    }
+
+    public function setAccount(string $account): void
+    {
+        $this->account = $account;
+        $this->iban = null;
+        $this->code = null;
+    }
+
+    public function getIban(): ?string
+    {
+        return $this->iban;
+    }
+
+    public function setIban(string $iban): void
+    {
+        $this->iban = $iban;
+        $this->account = null;
+        $this->code = null;
+    }
+
+    public function getCode(): ?string
+    {
+        return $this->code;
+    }
+
+    /**
+     * @param mixed $code
+     */
+    public function setCode($code): void
+    {
+        $this->code = $code;
+        $this->account = null;
+        $this->iban = null;
+    }
+
+    /**
+     * @return bool
+     */
+    public function isImportDuplicate(): bool
+    {
+        return $this->importDuplicate;
+    }
+
+    public function setImportDuplicate(bool $importDuplicate): void
+    {
+        $this->importDuplicate = $importDuplicate;
+    }
+
+    /**
+     * @return array|ElectronicBankStatementTransaction[]
+     */
+    public function getTransactions(): array
+    {
+        return $this->transactions;
+    }
+
+    /**
+     * @param array|ElectronicBankStatementTransaction[] $transactions
+     */
+    public function setTransactions(array $transactions): void
+    {
+        Assert::allIsInstanceOf($transactions, ElectronicBankStatementTransaction::class);
+        $this->transactions = $transactions;
+
+        foreach ($transactions as $transaction) {
+            $this->closevalue = $this->closevalue->add($transaction->getValue());
```

Comment:
This only works when `$this->closevalue` is already set to the start value in `setStartvalue()`. Maybe add an assertion that `$this->startvalue` is not empty?

And setting the transactions a second time will overwrite the transactions, but the close value won't be reset. I think a `$this->closevalue = $this->startvalue` before the foreach-loop will fix this.